sql/analyzer/resolve_natural_joins.go

```diff
@@ -179,3 +176,68 @@ func resolveNaturalJoins(ctx *sql.Context, a *Analyzer, n sql.Node) (sql.Node, e
 		})
 	})
 }
+
+func isOverTranformedNode(node sql.Node, transformed []*transformedJoin) (over bool, colsToUnresolve map[string][]string) {
+	node.TransformUp(func(n sql.Node) (sql.Node, error) {
```

Comment:
if you're not using the result, you should be using `plan.Inspect` instead (with which you can probably use `==` instead of `reflect.DeepEqual`